### PR TITLE
Hotfix/vehicle deserialization

### PR DESCRIPTION
**Pull request by Hoodathunk**
* Fixed error in `Vehicle` class where vehicle data was not being deserialized from Ship Monitor json.
* Added vehicle data testing in `TestShipMonitorDeserialization()`.

### CODE REVIEW

File: DataDefinitions/Vehicle.cs [hunk 1]
```diff
@@ -10,24 +10,44 @@ public class Vehicle
         public string loadout { get; private set; }
         public int rebuilds { get; private set; }
 
+        public string vehicleDefinition
+        {
+            get => vehicleDef.edname;
```
Comment (Tkael) [inline DataDefinitions/Vehicle.cs]: Needs null protection, as vehicleDef might be null.
`get => vehicleDef.edname;`

File: DataDefinitions/Vehicle.cs [hunk 2]
```diff
@@ -10,24 +10,44 @@ public class Vehicle
...
-        public string invariantName => definition?.invariantName;
+        public string invariantName => vehicleDef?.invariantName;
 
         [JsonIgnore]
         [Obsolete("Please be explicit and use localizedName or invariantName")]
         public string name => localizedName;
 
+        public string loadoutDescription
+        {
+            get => descriptionDef.edname;
```
Comment (Tkael) [inline DataDefinitions/Vehicle.cs]: Needs null protection, as descriptionDef might be null.
`get => descriptionDef.edname;`

File: DataDefinitions/Vehicle.cs (hunk 1 above)
Comment (Hoodathunk) [reply DataDefinitions/Vehicle.cs]: Fixed.

File: DataDefinitions/Vehicle.cs (hunk 2 above)
Comment (Hoodathunk) [reply DataDefinitions/Vehicle.cs]: Fixed.

**Pull request merged**